meshtastic/interfaces/ble/compatibility_service.py

```
@@ -23,6 +23,85 @@
+        self._iface = iface
+        self._publishing_thread_provider = publishing_thread_provider
+        self._last_publishing_thread: object | None = None
+
+    def _resolve_publishing_thread(self) -> object | None:
+        """Return publishing thread when available, else ``None`` on provider failure."""
+        try:
+            publishing_thread = self._publishing_thread_provider()
+            if publishing_thread is not None:
+                self._last_publishing_thread = publishing_thread
+                return publishing_thread
+            return self._last_publishing_thread
```

Ruff (0.15.6)

[warning] Consider moving this statement to an `else` block

(TRY300)